### PR TITLE
CA Renewal (Same Key Pair)

### PR DESCRIPTION
**Pull request by dangtony98**
# Description 📣

This PR adds CA renewal functionality that is the ability to renew the CA certificate of a CA; 

More specifically, it includes:
- Updates to CA, CA certificate and CA secret schemas to (1) Support one-to-many relationship between CA and CA certificates, (2) Keep track of the current version of CA certificate that should be considered active, (3) Keep track of which key-pair is linked to each CA certificate.
- New CA Page where users can view details about a CA and list of CA certificates; it's possible to download the CA certificate body and certificate chain for each.
- CA renewal with same key pair workflow; endpoints + UI.

More about the CA renewal workflow:
- If we are dealing with a Root CA, the same key pair is used to create a new self-signed certificate.
- If we are dealing with an Intermediate CA, the same key pair is used to generate a CSR; the private key of the parent CA is used against the CSR to generate the new certificate.

Note that currently, we only implement CA renewal with same key pair workflow but in the coming month will look to implement CA renewal with new key pair.

Screenshot 2024-08-05 at 8 15 43 PM

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Documentation

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

**Review by maidul98 — COMMENTED** [review]
Left some small comments. Looks pretty solid otherwise, nice work!

Note: Did not review frontend code. I think it would be better for Akhi or sheen to do that

File: backend/src/db/schemas/dynamic-secrets.ts [hunk 1]
```diff
@@ -14,16 +16,12 @@ export const DynamicSecretsSchema = z.object({
...
-  inputTag: z.string(),
-  algorithm: z.string().default("aes-256-gcm"),
-  keyEncoding: z.string().default("utf8"),
   folderId: z.string().uuid(),
   status: z.string().nullable().optional(),
   statusDetails: z.string().nullable().optional(),
   createdAt: z.date(),
-  updatedAt: z.date()
+  updatedAt: z.date(),
+  encryptedConfig: zodBuffer
```
[the remark this answers is not in the document]
Comment (dangtony98) [reply backend/src/db/schemas/dynamic-secrets.ts:24]: The `encryptedConfig` isn't a part of this PR; not sure why it's shown here.

File: backend/src/db/schemas/webhooks.ts [hunk 2]
```diff
@@ -5,27 +5,22 @@
...
 import { TImmutableDBKeys } from "./models";
 
 export const WebhooksSchema = z.object({
   id: z.string().uuid(),
   secretPath: z.string().default("/"),
-  url: z.string(),
   lastStatus: z.string().nullable().optional(),
   lastRunErrorMessage: z.string().nullable().optional(),
   isDisabled: z.boolean().default(false),
-  encryptedSecretKey: z.string().nullable().optional(),
```
[the remark this answers is not in the document]
Comment (dangtony98) [reply backend/src/db/schemas/webhooks.ts:17]: The `encryptedSecretKey` isn't a part of this PR; not sure why it's shown here.

File: backend/src/db/migrations/20240802181855_ca-cert-version.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+export async function up(knex: Knex): Promise<void> {
+  if (await knex.schema.hasTable(TableName.CertificateAuthority)) {
+    const hasActiveCaCertVersionColumn = await knex.schema.hasColumn(
+      TableName.CertificateAuthority,
+      "activeCaCertVersion"
+    );
+    if (!hasActiveCaCertVersionColumn) {
+      await knex.schema.alterTable(TableName.CertificateAuthority, (t) => {
+        t.integer("activeCaCertVersion").nullable();
```
[the remark this answers is not in the document]
Comment (dangtony98) [reply backend/src/db/migrations/20240802181855_ca-cert-version.ts]: Made these exact adjustments!

**Review by maidul98 — APPROVED** [review]

**Pull request merged**